splice_machine/src/test/java/com/splicemachine/subquery/Subquery_Flattening_InList_IT.java

```diff
@@ -360,4 +360,100 @@ public void numberOfColumnsMismatch() throws Exception {
             Assert.assertEquals("42X58", e.getSQLState());
         }
     }
+
+    @Test
+    public void testSetOpInSubQFlatten() throws Exception {
+
+        String expected =
+            "A1 |A2 |\n" +
+            "--------\n" +
+            " 0 | 0 |\n" +
+            " 1 |10 |\n" +
+            " 2 |20 |\n" +
+            " 3 |30 |\n" +
+            " 4 |40 |\n" +
+            " 5 |50 |";
+        String sql = "select a1,a2 from A\n" +
+                "WHERE a1 in (select a1 from A" +
+                "            WHERE a1 in (\n" +
+                "            SELECT b1 FROM B " +
+                "            UNION " +
+                "            SELECT b1 FROM B ))";
+        assertUnorderedResult(methodWatcher.getOrCreateConnection(),
+                              sql , ZERO_SUBQUERY_NODES, expected );
+        sql = "select a1,a2 from A\n" +
+                "WHERE a1 in (select a1 from A" +
+                "            WHERE a1+a1 in (\n" +
+                "            SELECT b1+b1 FROM B " +
+                "            UNION " +
+                "            SELECT b1 FROM B ))";
+        assertUnorderedResult(methodWatcher.getOrCreateConnection(),
+                              sql , ZERO_SUBQUERY_NODES, expected );
+        sql = "select a1,a2 from A\n" +
+                "WHERE a1 in (select a1 from A" +
+                "            WHERE a1+a1 in (\n" +
+                "            SELECT b1 FROM B " +
+                "            UNION " +
+                "            SELECT b1+b1 FROM B ))";
+        assertUnorderedResult(methodWatcher.getOrCreateConnection(),
+                              sql , ZERO_SUBQUERY_NODES, expected );
+
+        sql = "select a1,a2 from A\n" +
+                "WHERE exists (select a1 from A" +
+                "            WHERE a1+a1 in (\n" +
+                "            SELECT b1 FROM B " +
+                "            UNION " +
+                "            SELECT b1+b1 FROM B ))";
+        assertUnorderedResult(methodWatcher.getOrCreateConnection(),
+                              sql , ONE_SUBQUERY_NODE, expected );
```

Comment:
We have an IT called `Subquery_Flattening_Exists_Union_IT`, probably a better match for this case. But it's your decision whether to move it or not.

Reply:
OK, well I was mainly fixing this for IN list, so I put it there.  Also, this isn't specific to UNION, it's for all 3 set operation kinds.